build-tools/packages/build-cli/src/commands/modify/fluid-imports.ts

```diff
@@ -0,0 +1,334 @@
+		.filter((sourceFile) => sourceFile.getExtension() !== ".d.ts");
+
+	/**
+	 * List of source file save promises. Used to collect modified source file save promises so we can await them all at
+	 * once.
+	 */
+	const fileSavePromises: Promise<void>[] = [];
+
+	// Iterate over each source file, looking for Fluid imports
+	for (const sourceFile of sourceFiles) {
+		log?.verbose(`Source file: ${sourceFile.getBaseName()}`);
+
+		/**
+		 * All of the import declarations. This is basically every `import foo from bar` statement in the file.
+		 */
+		let imports = sourceFile.getImportDeclarations();
+
+		// Skip source files with no imports.
+		if (imports.length === 0) {
+			continue;
+		}
+
+		/**
+		 * True if the sourceFile has changed.
+		 */
+		let sourceFileChanged = false;
+
+		/**
+		 * A mapping of new module specifier to named import. We'll populate this as we scan the existing imports, then
+		 * write new remapped imports to the file.
+		 */
+		const newImports: Map<string, Set<string>> = new Map();
+
+		// First pass: collect the existing declarations
+		for (const importDeclaration of imports) {
+			// Skip non-Fluid imports
+			if (!isFluidImport(importDeclaration)) {
+				continue;
+			}
+			const [moduleName] = parseImport(importDeclaration);
+			const data = mappingData.get(moduleName);
+
+			// Skip modules with no mapping
+			if (data === undefined) {
+				log?.verbose(`Skipping ${importDeclaration.getModuleSpecifierValue()}`);
+			} else {
+				// TODO: Handle default import.
+				const defaultImport = importDeclaration.getDefaultImport();
+				if (defaultImport !== undefined) {
+					log?.warning(
+						`Found a default import (not yet implemented): ${defaultImport.getText().trim()}`,
+					);
+				}
+				const namedImports = importDeclaration.getNamedImports();
+
+				log?.logIndent(`Iterating named imports...`, 2);
+				for (const importSpecifier of namedImports) {
+					const name = importSpecifier.getName();
+
+					/**
+					 * fullImportSpecifierText includes surrounding text like "type" and whitespace. The surrounding whitespace is
+					 * trimmed, but leading or trailing text like "type" or "as foo" is still included. This is the string that
+					 * will be used in the new imports.
+					 *
+					 * This text also includes an alias if defined. Imports that are aliased are updated but the alias should not
+					 * be changed; they should remain the same after import cleanup, just possibly imported from a different
+					 * module/path.
+					 */
+					const fullImportSpecifierText = importSpecifier.getFullText().trim();
+					const expectedLevel = getApiLevelForImportName(name, data, "public", onlyInternal);
+
+					log?.logIndent(
+						`Found import named: '${fullImportSpecifierText}' (${expectedLevel})`,
+						4,
+					);
+					const newSpecifier =
+						expectedLevel === "public" ? moduleName : `${moduleName}/${expectedLevel}`;
```

Comment:
Which `"public"` strings are just `public` versus ones that `publicLevel`? We should use `publicLevel` if that is what these are.